system/autoware_default_adapi/src/planning.cpp

```diff
@@ -49,16 +49,117 @@ std::vector<typename rclcpp::Subscription<T>::SharedPtr> init_factors(
 }
 
 template <class T>
-T merge_factors(const rclcpp::Time stamp, const std::vector<typename T::ConstSharedPtr> & factors)
+std::vector<T> convert(const std::vector<PlanningFactor> & factors)
 {
-  T message;
+  static_assert(sizeof(T) == 0, "Only specializations of convert can be used.");
+  throw std::logic_error("Only specializations of convert can be used.");
+}
+
+template <>
+std::vector<VelocityFactor> convert(const std::vector<PlanningFactor> & factors)
+{
+  std::vector<VelocityFactor> velocity_factors;
+
+  for (const auto & factor : factors) {
+    if (factor.behavior != PlanningFactor::SLOW_DOWN && factor.behavior != PlanningFactor::STOP) {
+      continue;
+    }
+
+    if (factor.control_points.empty()) {
+      continue;
+    }
+
+    if (conversion_map.count(factor.module) == 0) {
+      continue;
+    }
+
+    VelocityFactor velocity_factor;
+    velocity_factor.behavior = conversion_map.at(factor.module);
+    velocity_factor.pose = factor.control_points.front().pose;
+    velocity_factor.distance = factor.control_points.front().distance;
+
+    velocity_factors.push_back(velocity_factor);
+  }
+
+  return velocity_factors;
+}
+
+template <>
+std::vector<SteeringFactor> convert(const std::vector<PlanningFactor> & factors)
+{
+  std::vector<SteeringFactor> steering_factors;
+
+  for (const auto & factor : factors) {
+    if (
+      factor.behavior != PlanningFactor::SHIFT_RIGHT &&
+      factor.behavior != PlanningFactor::SHIFT_LEFT &&
+      factor.behavior != PlanningFactor::TURN_RIGHT &&
+      factor.behavior != PlanningFactor::TURN_LEFT) {
+      continue;
+    }
+
+    if (factor.control_points.size() < 2) {
+      continue;
+    }
+
+    if (conversion_map.count(factor.module) == 0) {
+      continue;
+    }
+
```

Comment:
Please also add a check for `direction_map` to avoid an exception by the `at` function.

Reply:
Fixed in https://github.com/tier4/autoware.universe/pull/1734/commits/615d1bb7371048c3c0bf5a71f067a92e378c8d9e.